### PR TITLE
Added integration with the image-builder kubevirt image

### PR DESCRIPTION
**Pull request by isaacdorfman**
**What this PR does / why we need it**:
I created a kubevirt target in my fork of the image-builder project:
https://github.com/isaacdorfman/image-builder/tree/kubevirt-target

This image is the image of the machines in the cluster.
I pushed the image I built in my image-builder fork into quay and updated the capk to use this image instead of the kubevirtci one.
In this PR I also fixed the problems that this change caused.
*I also fixed a problem with the local container reigstry by changing the url of the registry from localhost to 127.0.0.1, this caused problems related to ipv6 in some devices.
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release notes**:

```release-note
```

### CODE REVIEW

**Comment by davidvossel** [comment]
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubernetes-sigs_cluster-api-provider-kubevirt/108/pull-kubernetes-sigs-cluster-api-provider-kubevirt-e2e/1501230590018457600

```
• Failure [620.180 seconds]
CreateCluster
/home/prow/go/src/github.com/kubernetes-sigs/cluster-api-provider-kubevirt/e2e-tests/create-cluster_test.go:30
  creating a simple cluster with persistent VMs [It]
  /home/prow/go/src/github.com/kubernetes-sigs/cluster-api-provider-kubevirt/e2e-tests/create-cluster_test.go:361
  Timed out after 600.000s.
      
  cluster should have control plane initialized
  Expected success, but got an error:
      <*errors.errorString | 0xc000913130>: {
          s: "still waiting on controlPlaneInitialized condition to be true",
      }
      still waiting on controlPlaneInitialized condition to be true
  /home/prow/go/src/github.com/kubernetes-sigs/cluster-api-provider-kubevirt/e2e-tests/create-cluster_test.go:257
------------------------------

```
It looks like the control plane didn't start up successfully using this image.

**Comment by davidvossel** [comment]
ah, but it was only the `creating a simple cluster with persistent VMs [It]` test that failed. The others passed! so that's a great initial starting point. Just investigate why that one test ```creating a simple cluster with persistent VMs [It]``` is failing. Maybe you can reproduce this locally?

File: Makefile
Comment (isaacdorfman) [inline Makefile:59]: This fixes an issue with operating systems that translate dns names to ipv6.

File: e2e-tests/create-cluster_test.go
Comment (nirarg) [inline e2e-tests/create-cluster_test.go:315]: Need to change to the official registry before merge

File: config/default/manager_image_patch.yaml
Comment (nirarg) [inline config/default/manager_image_patch.yaml:11]: I'm not what effects this would have
@agradouski @cchengleo is this fine by you to change it permanently?

File: Makefile
Comment (nirarg) [reply Makefile:59]: This is fixing issue @isaacdorfman has on his computer
Is it ok changing it here in order to avoid similar errors in the future?
@davidvossel @agradouski @cchengleo @nunnatsa  ^

File: e2e-tests/create-cluster_test.go
Comment (nirarg) [inline e2e-tests/create-cluster_test.go:99]: What does this code test?
This is even not failing the test in case of error

File: go.mod
Comment (nirarg) [inline go.mod:23]: Please remove this change, as it does nothing

File: kubevirtci
Comment (nirarg) [inline kubevirtci:101]: Can you please elaborate, why was this changed?

File: Makefile
Comment (davidvossel) [reply Makefile:59]: i think it's likely fine

Comment (cchengleo) [reply Makefile:59]: I think this will be okay as we always specify the REGISTRY env variable.

File: kubevirtci
Comment (isaacdorfman) [reply kubevirtci:101]: The new image uses containerd instead of cri-o.
And containerd listens on the "/var/run/containerd/containerd.sock" socket.

File: e2e-tests/create-cluster_test.go
Comment (isaacdorfman) [reply e2e-tests/create-cluster_test.go]: This is was used for my debug, I removed it completely now.

File: config/default/manager_image_patch.yaml
Comment (isaacdorfman) [reply config/default/manager_image_patch.yaml]: Done

File: e2e-tests/create-cluster_test.go
Comment (isaacdorfman) [reply e2e-tests/create-cluster_test.go]: Done, but 20.04 instead of 18.04 because that's the image that was used.

File: Makefile
Comment (agradouski) [reply Makefile:59]: should be fine

File: kubevirtci
Comment (agradouski) [reply kubevirtci:101]: 👍

**Comment by davidvossel** [comment]
PR looks great, however when i tested this locally, the persistent storage e2e test failed.

I checked out this PR and did some digging. The issue is that this new ubuntu image expands to be 20GI, and the local storage provisioner isn't allowing us to import it successfully.

I was able to get this working though by switching us to use rook-ceph as the storage class. We'll need #127 to go in order for this to pass the entire e2e suite.

**Comment by isaacdorfman** [comment]
@davidvossel I cherry-picked the commit and the CI succeeded.

**Comment by davidvossel** [comment]
/hold cancel

sorry, you'll have to rebase. my pr went in

**Comment by davidvossel** [comment]
this PR https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/pull/126  will impact you as well. Make sure to alter the env vars that are moved to hack/functest.sh

**Comment by davidvossel** [comment]
/hold 

waiting for ci to pass

File: e2e/create-cluster_test.go
Comment (davidvossel) [inline e2e/create-cluster_test.go:440]: these env changes need to be modified in `hack/functest.sh` rather than being in the e2e test

**Review by davidvossel — COMMENTED** [review]
/lgtm cancel

**Review by davidvossel — APPROVED** [review]
/lgtm

**Pull request merged**
